src/contours.js

```diff
@@ -167,18 +172,23 @@ export default function() {
           xt = x | 0,
           yt = y | 0,
           v0,
-          v1 = values[yt * dx + xt];
-      if (x > 0 && x < dx && xt === x) {
-        v0 = values[yt * dx + xt - 1];
+          v1 = low(values[yt * dx + xt]);
+        if (x > 0 && x < dx && xt === x) {
+        v0 = low(values[yt * dx + xt - 1]);
+        if (v0 == null || isNaN(v0) || !isFinite(v0)) v0 = -1e52;
```

Comment:
This…

* is redundant with `low(…)`
* treats positive infinity as negative infinity
* uses a magic number when we can handle infinity exactly

The passed-in `value` should be coerced like this (outside this function, probably in `contour` as I mentioned in the other comment):

```js
value = value === null ? NaN : +value;
if (isNaN(value)) throw new Error(`invalid value: ${value}`);
```

Values we read from `values` should be coerced like this:

```js
v0 = v0 === null ? NaN : +v0;
v1 = v1 === null ? NaN : +v1;
```

Given that, maybe something like this?

```js
const a = value - v0;
const b = v1 - v0;
const d = isFinite(a) && isFinite(b) ? a / b : Math.sign(a) / Math.sign(b);
point[0] = isNaN(d) ? x : x + d - 0.5;
```